### PR TITLE
[new release] carton, carton-lwt and carton-git (0.1.0)

### PR DESCRIPTION
**Pull request by dinosaure**
CHANGES:

- First release of carton

Close #17941

### CODE REVIEW

File: packages/carton-git/carton-git.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+]
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+dev-repo: "git+https://github.com/dinosaure/carton.git"
+x-commit-hash: "d7d3ddfccf0aa2c3f273481ec2ef5b84f3a4d054"
+url {
+  src:
+    "https://github.com/dinosaure/carton/releases/download/carton-v0.1.0/carton-carton-v0.1.0.tbz"
```
Comment (kit-ty-kate) [inline packages/carton-git/carton-git.0.1.0/opam:44]: Is it expected for `carton-git` and `carton-lwt` to have a different archive from `carton`? (`dinosaure/carton` vs. `mirage/ocaml-git`)

**Comment by kit-ty-kate** [comment]
`carton-lwt`'s tests failed:
```
#=== ERROR while compiling carton-lwt.0.1.0 ===================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.11.1 | pinned(https://github.com/dinosaure/carton/releases/download/carton-v0.1.0/carton-carton-v0.1.0.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/carton-lwt.0.1.0
# command              ~/.opam/4.11/bin/dune runtest -p carton-lwt -j 47
# exit-code            1
# env-file             ~/.opam/log/carton-lwt-15-bcdb80.env
# output-file          ~/.opam/log/carton-lwt-15-bcdb80.out
### output ###
#         test alias test/carton/runtest (exit 1)
# (cd _build/default/test/carton && ./test.exe --color=always)
# Initialized empty Git repository in /tmp/git-0a7022/.git/
# Initialized empty Git repository in /tmp/git-ba5401/.git/
# pack	da39a3ee5e6b4b0d3255bfef95601890afd80709
# Testing `carton'.
# This run has ID `0CE2D6E4-F17F-4133-AC39-4C4484294489'.
# 
#   [OK]          weights          0   weight.
#   [OK]          loads            0   load.
#   [OK]          decoder          0   valid empty pack.
#   [OK]          decoder          1   verify empty pack.
#   [OK]          decoder          2   check empty index.
#   [OK]          decoder          3   verify & generate index of bomb pack.
#   [OK]          decoder          4   first entry of bomb pack.
#   [OK]          decoder          5   unpack bomb pack.
#   [OK]          decoder          6   decode index stream.
#   [OK]          decoder          7   empty stream.
#   [OK]          encoder          0   empty pack.
#   [OK]          encoder          1   index of empty pack.
#   [OK]          encoder          2   index of one entry.
# > [FAIL]        encoder          3   pack bomb pack.
#   [OK]          encoder          4   cycle.
#   [OK]          lwt              0   map-yield.
# 
# ┌──────────────────────────────────────────────────────────────────────────────┐
# │ [FAIL]        encoder          3   pack bomb pack.                           │
# └──────────────────────────────────────────────────────────────────────────────┘
# ASSERT new.pack
# ASSERT hash
# FAIL hash
# 
#    Expected: `f1c028c3febcef55dc93d41ff4715d819d6a55da'
#    Received: `994fa34a28e8cf38ca8438674b66ded0b86f2704'
# 
# Logs saved to `~/.opam/4.11/.opam-switch/build/carton-lwt.0.1.0/_build/default/test/carton/_build/_tests/carton/encoder.003.output'.
#  ──────────────────────────────────────────────────────────────────────────────
# 
# Full test results in `~/.opam/4.11/.opam-switch/build/carton-lwt.0.1.0/_build/default/test/carton/_build/_tests/carton'.
# 1 failure! in 0.000s. 16 tests run.
```

**Comment by dinosaure** [comment]
> carton-lwt's tests failed:

Yes, I don't really understand why. It's seems unrelated to the scheduler and I'm not able to reproduce that on my machine (and I don't have Oracle 7 or CentOS 7 on my side). So I don't know what to do to fix that 😕 .

File: packages/carton-git/carton-git.0.1.0/opam (hunk 1 above)
Comment (dinosaure) [reply packages/carton-git/carton-git.0.1.0/opam:44]: Seems related to the OPAM file, I will fix that.

**Comment by kit-ty-kate** [comment]
> (and I don't have Oracle 7 or CentOS 7 on my side). So I don't know what to do to fix that confused .

You can use the docker images: `docker run --rm -it ocaml/opam:centos-7`

**Comment by dinosaure** [comment]
With a big introspection, it seems that `git.1.8.5.3` (CentOS 7 and Oracle Linux 7) has a bug on the `git-index-pack` binary. `git.1.9.0` does not have the same bug and it seems fixed then but `ocaml-git` does what it should does - and produce the right hash. I added a condition on this specific test to run only if the version of Git is superior or equal to 1.9.0. With that, we should have a green CI.

However, I suspect a real bug on Git and we should make an issue or send something to the Git team (even if it's an old version).

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Pull request merged**